src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -417,6 +424,135 @@ bool emberAfOperationalCredentialsClusterUpdateNOCCallback(EndpointId endpoint,
     return true;
 }
 
+bool emberAfOperationalCredentialsClusterCertChainRequestCallback(EndpointId endpoint, app::CommandHandler * commandObj,
+                                                                  uint16_t certChainType)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    CHIP_ERROR err       = CHIP_NO_ERROR;
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: commissioner has requested Device Attestation Credentials");
+
+    app::CommandPathParams cmdParams = { emberAfCurrentEndpoint(), /* group id */ 0, ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID,
+                                         ZCL_CERT_CHAIN_RESPONSE_COMMAND_ID, (app::CommandPathFlags::kEndpointIdValid) };
+
+    TLV::TLVWriter * writer = nullptr;
+    uint8_t derBuf[Credentials::kMaxDERCertLength];
+    MutableByteSpan derBufSpan(derBuf);
+
+    // TODO: remove line below
+    Credentials::SetDeviceAttestationCredentialsProvider(Credentials::Examples::GetExampleDACProvider());
+    Credentials::DeviceAttestationCredentialsProvider * dacProvider = Credentials::GetDeviceAttestationCredentialsProvider();
+
+    VerifyOrExit(commandObj != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    SuccessOrExit(err = commandObj->PrepareCommand(cmdParams));
+    writer = commandObj->GetCommandDataElementTLVWriter();
+    if (certChainType == kDACCertificate)
+    {
+        SuccessOrExit(err = dacProvider->GetDeviceAttestationCert(derBufSpan));
+    }
+    else if (certChainType == kPAICertificate)
+    {
+        SuccessOrExit(err = dacProvider->GetProductAttestationIntermediateCert(derBufSpan));
+    }
+    else
+    {
+        SuccessOrExit(status = EMBER_ZCL_STATUS_FAILURE);
+    }
+    SuccessOrExit(err = writer->Put(TLV::ContextTag(0), derBufSpan));
```

Comment:
This is the incorrect tag ID for the response:

```
[width="99%",frame="all",options="header", cols="^,<s,<,^,^"]
|===
| Id | Field            | Type                    | Constraint    | Conformance
| 1  | Certificate      | <<octstr-def, octstr>>  | max 600       | M
|===
```

Let's set the ContextTag to 1 in both the Client and Server until we resolve whether the spec should use 0 here.

Reply:
IMClusterrCommandHandler.cpp (zap auto generated file) always parses TLVs starting with TLV Tag ID 0. So we cannot change it

Reply:
If ZAP is not spec compliant, then we have to fix it. In the meantime, please add a TODO here.